src/liquid/components/ld-select/ld-select.tsx

```diff
@@ -630,63 +691,65 @@ export class LdSelect implements InnerFocusable {
   private handleEnd = (ev) => {
     // Move focus to the last option.
     ev.preventDefault()
-    const options = Array.from(
+    const visibleOptions = Array.from(
       this.listboxRef.querySelectorAll('ld-option-internal')
-    )
-    if (document.activeElement !== options[options.length - 1]) {
-      options[options.length - 1].focusOption()
+    ).filter((option) => !isLdOptInternalHidden(option))
+    if (document.activeElement !== visibleOptions[visibleOptions.length - 1]) {
+      visibleOptions[visibleOptions.length - 1].focusInner()
     }
   }
 
   private selectAndFocus = (
     ev: KeyboardEvent,
-    ldOption: HTMLLdOptionInternalElement
+    opt: HTMLLdOptionInternalElement | HTMLLdOptgroupInternalElement | undefined
```

Comment:
I'll keep this as it is: The opt is not optional, there is always a nextElementSibling or a previousElementSibling that must be passed to the function, but it can be `undefined`.